### ISSUE TITLE
Annotations not reflecting test outcome with jest@23.0.0-alpha.0

### ISSUE THREAD

**Issue by TrevorBurnham** [jest-community/vscode-jest#282]
### Environment

1. `node -v`: v8.10.0
2. `npm -v`: 5.7.1

4. Operating system: macOS 10.13.3

### Steps to Reproduce

The issue appears to occur in any project running `jest@23.0.0-alpha.0`. I've created a simple repo to demonstrate:

```sh
> git clone https://github.com/TrevorBurnham/jest-23-bug-report
> code jest-23-bug-report
```

I'm running the Jest extension version 2.6.4 under VSCode 1.21.1.

### Expected Behavior

The extension shows a filled-in green dot next to the sole test, indicating that it passes.

### Actual Behavior

The extension shows a hollow green dot next to the sole test:

![image](https://user-images.githubusercontent.com/224895/37492138-97c36dcc-2876-11e8-890d-546eef6ae28d.png)

The same annotation is also shown for tests that fail.

**Comment by seanpoulter** [comment]
What does the full output from the Jest channel say @TrevorBurnham? I can reproduce the issue but with the example repo there aren't any changes in the repo for watch mode to run so it is sitting waiting for a message indicating that "no tests found since last commit" ...:

<img width="1136" alt="screen shot 2018-03-15 at 22 37 58" src="https://user-images.githubusercontent.com/2585460/37500762-ac20930a-28a1-11e8-8760-bfd7affebd33.png">

**Comment by seanpoulter** [comment]
The extension will run all the tests once, then start up in watch mode. The issue seems to be caused by the extension not detecting when `jest-editor-support` tells us that the `"No tests found related to files changed since last commit."` message has been recognized.

Would you like to try fixing it? The test runner in `jest-editor-support` _should_ already be detecting this and returning more information like an "exit code" from this block: https://github.com/facebook/jest/blob/master/packages/jest-editor-support/src/Runner.js#L150

**Comment by seanpoulter** [comment]
As of the Jest 23 alpha the `"Test results written to"` message is no longer written to `stdout`. This causes the `Runner` in `jest-editor-support` not to know where the results have been saved and the `executableJSON` event not to fire.

**Comment by utillity** [comment]
correct! I just refactored the couple of lines of code from ...on('data'... to a separate method and call the method when data is received from stdout and stderr and now it works

	checkIfResultsFilePopulated(data) {
		// Make jest save to a file, otherwise we get chunked data
		// and it can be hard to put it back together.
		const stringValue = data.
		toString().
		replace(/\n$/, '').
		trim();

		if (stringValue.startsWith('Test results written to')) {
			(0, (_fs || _load_fs()).readFile)(this.outputPath, 'utf8', (err, data) => {
				if (err) {
					const message = `JSON report not found at ${this.outputPath}`;
					this.emit('terminalError', message);
				} else {
					const noTestsFound = this.doResultsFollowNoTestsFoundMessage();
					this.emit('executableJSON', JSON.parse(data), { noTestsFound });
				}
			});
		} else {
			this.emit('executableOutput', stringValue.replace('[2J[H', ''));
		}
	}
actually since you're not trying to parse the filename anyways, why not use a file-watcher and act on it when the file has been created or changed?
